metrics/resource_view.go

```diff
@@ -51,14 +54,51 @@ type meters struct {
 	meters  map[string]*meterExporter
 	factory ResourceExporterFactory
 	lock    sync.Mutex
+	clock   clock.Clock
+	ticker  clock.Ticker
 }
 
 // Lock regime: lock allMeters before resourceViews. The critical path is in
 // optionForResource, which must lock allMeters, but only needs to lock
 // resourceViews if a new meter needs to be created.
-var resourceViews = storedViews{}
-var allMeters = meters{
-	meters: map[string]*meterExporter{"": &defaultMeter},
+var (
+	resourceViews = storedViews{}
+	allMeters     = meters{
+		meters: map[string]*meterExporter{"": &defaultMeter},
+		clock:  cleanupClock,
+	}
+
+	cleanupClock               = clock.Clock(clock.RealClock{}) // Var needed to avoid allMeters->defaultMeter->allMeters.clock cycle
```

Comment:
Is this still true? Can't you just initialize the clock in `allMeters` with this now?

Reply:
See comment on defaultMeter initialization.  The meters need a timestamp, but for defaultMeter the value doesn't matter since it never expires (so I'll just avoid reading the timestamp in that case).

I'll remove this.